packages/next/next-server/lib/router/router.ts

```diff
@@ -316,22 +316,32 @@ export default class Router implements BaseRouter {
 
       if (isDynamicRoute(route)) {
         const { pathname: asPathname } = parse(as)
-        const routeMatch = getRouteMatcher(getRouteRegex(route))(asPathname)
+        const routeRegex = getRouteRegex(route)
+        const routeMatch = getRouteMatcher(routeRegex)(asPathname)
         if (!routeMatch) {
-          const error =
-            `The provided \`as\` value (${asPathname}) is incompatible with the \`href\` value (${route}). ` +
-            `Read more: https://err.sh/zeit/next.js/incompatible-href-as`
-
           if (process.env.NODE_ENV !== 'production') {
-            throw new Error(error)
-          } else {
-            console.error(error)
+            console.warn(
+              `The provided \`as\` value (${asPathname}) is incompatible with the \`href\` value (${route}). ` +
+                `Read more: https://err.sh/zeit/next.js/incompatible-href-as`
+            )
+
+            const missingParams = Object.keys(routeRegex.groups).filter(
+              param => !query[param]
+            )
+
+            if (missingParams.length > 0) {
```

Comment:
It looks like we're no longer blocking navigation in prod like we used to. We should probably still throw an error in production so it renders `/_error`.